### ISSUE TITLE
GhQuantities test fails for specific seed 1396486037

### ISSUE THREAD

**Issue by nilsdeppe** [sxs-collaboration/spectre#1392]
# Bug reports:

```
Charm++: standalone mode (not using charmrun)
Charm++> Running in Multicore mode:  1 threads
Converse/Charm++ Commit ID: v6.8.0-0-ga36028edb
Warning> Randomization of virtual memory (ASLR) is turned on in the kernel, thread migration may not work! Run 'echo 0 > /proc/sys/kernel/randomize_va_space' as root to disable it, or try running with '+isomalloc_sync'.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 unique compute nodes (2-way SMP).
Charm++> cpu topology info is gathered in 0.000 seconds.
SpECTRE Build Information:
Version:                      0.0.0
Compiled on host:             cb108458b265
Compiled in directory:        /work/build_Release_clang
Source directory is:          /work/spectre
Compiled on git branch:       HEAD
Compiled with git hash:       6fafef586722808864e04e9fdebe0e0b838f30cb
Linked on:                    Mon Feb 25 00:07:56 2019
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
"Release.Unit.PointwiseFunctions.GeneralRelativity.GhQuantities" is a Catch v2.1.0 host application.
Run with -? for options
-------------------------------------------------------------------------------
Unit.PointwiseFunctions.GeneralRelativity.GhQuantities
-------------------------------------------------------------------------------
/work/spectre/tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeGhQuantities.cpp:514
...............................................................................
/work/spectre/tests/Unit/TestingFramework.hpp:137: FAILED:
  CHECK( a == appx(b) )
with expansion:
  -110.4440092629
  ==
  Approx( -110.4440092615 )
with messages:
  seed: 1396486037
  function: spacetime_deriv_norm_shift
  /work/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:222: result == pypp::
  call<ResultType>( module_name, function_name, std::get<ArgumentIs>(args)...,
  forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
  used_for_size)...)
  a: T(0)=(2.810831418551998213e+06,-1.104440092628647108e+02,-1.
  299604505127133802e+06,1.586896518673141953e+06,5.221432038981864229e+06,3.
  312903229346902226e+05,5.136470857733141165e+05,6.175607965452162549e+06,2.
  588695034359634388e+06,7.517181165573709877e+05,6.146523471216153121e+05,3.
  927980362669613212e+06,8.183837586696650833e+05,9.118104651397459675e+05,2.
  589767780696331523e+06,1.152537022488375951e+05,9.103222824844755232e+05,1.
  749319201215901878e+06,1.371438480238805059e+06,-2.151669390394785441e+06)
  T(1)=(1.868950780497653410e+06,4.459765874629504047e+06,9.
  362173754328757059e+05,1.506230499050474260e+06,3.791328691886513494e+06,2.
  666743828527465812e+05,5.816501356199600268e+05,5.075952309542067349e+06,1.
  151214396654948592e+06,2.734197244339024648e+06,4.211382775781658711e+05,6.
  778744064101103693e+06,1.382761545229653362e+06,2.310065923120175023e+06,7.
  056433727635389660e+05,1.355764147718349937e+06,4.221466445858031511e+05,5.
  755223401360389544e+05,2.940389955868895166e+06,2.114329894304316863e+07)
  T(2)=(2.377329011917619966e+06,5.948230894318724051e+06,1.
  205345092756808270e+06,1.165779717940017115e+06,6.800310989165732637e+06,1.
  617297277204716229e+05,4.067630067057749839e+05,4.332525772814889438e+06,2.
  040759301758227870e+06,3.130388876251514535e+06,8.178605974241105141e+05,5.
  107990465531982481e+06,1.099678033786474029e+06,4.295392037029253319e+06,1.
  023269826308578369e+06,1.164399460383182392e+06,3.650200210773987928e+05,5.
  195891105114031234e+05,3.409630895069878083e+06,9.669771987836781889e+06)
  b: T(0)=(2.810831418551997282e+06,-1.104440092614568130e+02,-1.
  299604505127133802e+06,1.586896518673141254e+06,5.221432038981864229e+06,3.
  312903229346898734e+05,5.136470857733141165e+05,6.175607965452163480e+06,2.
  588695034359634388e+06,7.517181165573709877e+05,6.146523471216151956e+05,3.
  927980362669615541e+06,8.183837586696653161e+05,9.118104651397458510e+05,2.
  589767780696331058e+06,1.152537022488376242e+05,9.103222824844754068e+05,1.
  749319201215901412e+06,1.371438480238805292e+06,-2.151669390394787304e+06)
  T(1)=(1.868950780497653643e+06,4.459765874629504979e+06,9.
  362173754328757059e+05,1.506230499050474027e+06,3.791328691886513494e+06,2.
  666743828527465812e+05,5.816501356199596776e+05,5.075952309542065486e+06,1.
  151214396654948592e+06,2.734197244339023717e+06,4.211382775781659293e+05,6.
  778744064101103693e+06,1.382761545229653595e+06,2.310065923120175023e+06,7.
  056433727635389660e+05,1.355764147718349937e+06,4.221466445858032093e+05,5.
  755223401360389544e+05,2.940389955868895631e+06,2.114329894304317236e+07)
  T(2)=(2.377329011917619500e+06,5.948230894318724051e+06,1.
  205345092756808270e+06,1.165779717940017115e+06,6.800310989165732637e+06,1.
  617297277204716229e+05,4.067630067057749256e+05,4.332525772814889438e+06,2.
  040759301758227870e+06,3.130388876251515001e+06,8.178605974241105141e+05,5.
  107990465531981550e+06,1.099678033786474261e+06,4.295392037029255182e+06,1.
  023269826308578253e+06,1.164399460383182392e+06,3.650200210773987928e+05,5.
  195891105114031234e+05,3.409630895069878083e+06,9.669771987836783752e+06)
  a: (2.810831418551998213e+06,-1.104440092628647108e+02,-1.
  299604505127133802e+06,1.586896518673141953e+06,5.221432038981864229e+06,3.
  312903229346902226e+05,5.136470857733141165e+05,6.175607965452162549e+06,2.
  588695034359634388e+06,7.517181165573709877e+05,6.146523471216153121e+05,3.
  927980362669613212e+06,8.183837586696650833e+05,9.118104651397459675e+05,2.
  589767780696331523e+06,1.152537022488375951e+05,9.103222824844755232e+05,1.
  749319201215901878e+06,1.371438480238805059e+06,-2.151669390394785441e+06)
  b: (2.810831418551997282e+06,-1.104440092614568130e+02,-1.
  299604505127133802e+06,1.586896518673141254e+06,5.221432038981864229e+06,3.
  312903229346898734e+05,5.136470857733141165e+05,6.175607965452163480e+06,2.
  588695034359634388e+06,7.517181165573709877e+05,6.146523471216151956e+05,3.
  927980362669615541e+06,8.183837586696653161e+05,9.118104651397458510e+05,2.
  589767780696331058e+06,1.152537022488376242e+05,9.103222824844754068e+05,1.
  749319201215901412e+06,1.371438480238805292e+06,-2.151669390394787304e+06)
  a: -1.104440092628647108e+02
  b: -1.104440092614568130e+02
```

**Comment by fmahebert** [comment]
In PR #1433, this test failed with seed `3093017092`.

Here's the full output:
```
/work/spectre/tests/Unit/TestingFramework.hpp:137: FAILED:
  CHECK( a == appx(b) )
with expansion:
  -11.1432766537 == Approx( -11.1432767124 )
with messages:
  seed: 3093017092
  function: dt_lower_shift
  /work/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:223: result == pypp::
  call<ResultType>( module_name, function_name, std::get<ArgumentIs>(args)...,
  forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
  used_for_size)...)
  a: T(0)=(4.694530637980111351e+03,2.101003495459987789e+03,3.
  207957317489257548e+03,1.393483476852223066e+04,3.629355746497701148e+03,-3.
  122852691247762777e+03,7.947428304515296986e+03,4.667399489592389727e+03,-1.
  305067346967442063e+03,-5.265219897160022811e+00,6.563177097183955084e+03,-1.
  114327665371820331e+01,3.737487417360972586e+02,2.074052821545474671e+03,3.
  957227271612302502e+03,1.778574118834093952e+03,1.504700200689648682e+03,1.
  784741275116078214e+03,1.626117516992070250e+03,1.463949266660182184e+04)
  T(1)=(4.884671151043688951e+03,2.016477673935037501e+03,2.
  563960072230777314e+03,8.422867877002194291e+03,3.971862617510039854e+03,-3.
  649419652202172074e+03,5.318344563287952042e+03,7.922373539234847158e+03,-2.
  036950370671185738e+03,-8.802316065768445696e+02,7.166888885237684917e+03,4.
  395276807617396116e+03,4.327169640156197374e+02,2.596848396524968848e+03,2.
  326512227834610712e+03,1.730263206899515808e+03,1.488983722766563233e+03,3.
  220665385083842466e+03,1.106893387043291796e+03,1.744565244349394197e+04)
  T(2)=(4.591014859591384266e+03,1.939544184320808199e+03,2.
  976166818405885351e+03,1.301132024216321406e+04,4.087264049876992431e+03,-3.
  398034635015842468e+03,6.939880618995272016e+03,8.703983955028797936e+02,-3.
  444151928813199447e+03,4.665937509833509466e+03,6.219091754047957693e+03,-7.
  533198408782482147e+00,7.007979966733214496e+02,2.162273754403387557e+03,3.
  659841872412106568e+03,1.167389104634420846e+03,1.116522333940990393e+03,-7.
  243475142061594170e+02,1.154319035451656646e+03,1.494881545793028999e+04)
  b: T(0)=(4.694530637980111351e+03,2.101003495459988244e+03,3.
  207957317489257093e+03,1.393483476852222884e+04,3.629355746497698419e+03,-3.
  122852691247709117e+03,7.947428304515295167e+03,4.667399489592389727e+03,-1.
  305067346967445701e+03,-5.265219897161060203e+00,6.563177097183949627e+03,-1.
  114327671237384720e+01,3.737487417360972017e+02,2.074052821545472398e+03,3.
  957227271612303866e+03,1.778574118834093042e+03,1.504700200689648454e+03,1.
  784741275116079123e+03,1.626117516992070250e+03,1.463949266660182548e+04)
  T(1)=(4.884671151043686223e+03,2.016477673935037728e+03,2.
  563960072230776859e+03,8.422867877002190653e+03,3.971862617510039399e+03,-3.
  649419652202167981e+03,5.318344563287955680e+03,7.922373539234846248e+03,-2.
  036950370671189603e+03,-8.802316065768432054e+02,7.166888885237679460e+03,4.
  395276807519043359e+03,4.327169640156197943e+02,2.596848396524966574e+03,2.
  326512227834611167e+03,1.730263206899513989e+03,1.488983722766563687e+03,3.
  220665385083842921e+03,1.106893387043292933e+03,1.744565244349394925e+04)
  T(2)=(4.591014859591383356e+03,1.939544184320808199e+03,2.
  976166818405885351e+03,1.301132024216320497e+04,4.087264049876989702e+03,-3.
  398034635015794720e+03,6.939880618995275654e+03,8.703983955028777473e+02,-3.
  444151928813204449e+03,4.665937509833510376e+03,6.219091754047954055e+03,-7.
  533198432627855823e+00,7.007979966733214496e+02,2.162273754403384828e+03,3.
  659841872412107477e+03,1.167389104634420391e+03,1.116522333940990393e+03,-7.
  243475142061581664e+02,1.154319035451656873e+03,1.494881545793030273e+04)
  a: (4.694530637980111351e+03,2.101003495459987789e+03,3.207957317489257548e+
  03,1.393483476852223066e+04,3.629355746497701148e+03,-3.122852691247762777e+
  03,7.947428304515296986e+03,4.667399489592389727e+03,-1.305067346967442063e+
  03,-5.265219897160022811e+00,6.563177097183955084e+03,-1.114327665371820331e+
  01,3.737487417360972586e+02,2.074052821545474671e+03,3.957227271612302502e+
  03,1.778574118834093952e+03,1.504700200689648682e+03,1.784741275116078214e+
  03,1.626117516992070250e+03,1.463949266660182184e+04)
  b: (4.694530637980111351e+03,2.101003495459988244e+03,3.207957317489257093e+
  03,1.393483476852222884e+04,3.629355746497698419e+03,-3.122852691247709117e+
  03,7.947428304515295167e+03,4.667399489592389727e+03,-1.305067346967445701e+
  03,-5.265219897161060203e+00,6.563177097183949627e+03,-1.114327671237384720e+
  01,3.737487417360972017e+02,2.074052821545472398e+03,3.957227271612303866e+
  03,1.778574118834093042e+03,1.504700200689648454e+03,1.784741275116079123e+
  03,1.626117516992070250e+03,1.463949266660182548e+04)
  a: -1.114327665371820331e+01
  b: -1.114327671237384720e+01

/work/spectre/tests/Unit/TestingFramework.hpp:137: FAILED:
  CHECK( a == appx(b) )
with expansion:
  -7.5331984088 == Approx( -7.5331984326 )
with messages:
  seed: 3093017092
  function: dt_lower_shift
  /work/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:223: result == pypp::
  call<ResultType>( module_name, function_name, std::get<ArgumentIs>(args)...,
  forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
  used_for_size)...)
  a: T(0)=(4.694530637980111351e+03,2.101003495459987789e+03,3.
  207957317489257548e+03,1.393483476852223066e+04,3.629355746497701148e+03,-3.
  122852691247762777e+03,7.947428304515296986e+03,4.667399489592389727e+03,-1.
  305067346967442063e+03,-5.265219897160022811e+00,6.563177097183955084e+03,-1.
  114327665371820331e+01,3.737487417360972586e+02,2.074052821545474671e+03,3.
  957227271612302502e+03,1.778574118834093952e+03,1.504700200689648682e+03,1.
  784741275116078214e+03,1.626117516992070250e+03,1.463949266660182184e+04)
  T(1)=(4.884671151043688951e+03,2.016477673935037501e+03,2.
  563960072230777314e+03,8.422867877002194291e+03,3.971862617510039854e+03,-3.
  649419652202172074e+03,5.318344563287952042e+03,7.922373539234847158e+03,-2.
  036950370671185738e+03,-8.802316065768445696e+02,7.166888885237684917e+03,4.
  395276807617396116e+03,4.327169640156197374e+02,2.596848396524968848e+03,2.
  326512227834610712e+03,1.730263206899515808e+03,1.488983722766563233e+03,3.
  220665385083842466e+03,1.106893387043291796e+03,1.744565244349394197e+04)
  T(2)=(4.591014859591384266e+03,1.939544184320808199e+03,2.
  976166818405885351e+03,1.301132024216321406e+04,4.087264049876992431e+03,-3.
  398034635015842468e+03,6.939880618995272016e+03,8.703983955028797936e+02,-3.
  444151928813199447e+03,4.665937509833509466e+03,6.219091754047957693e+03,-7.
  533198408782482147e+00,7.007979966733214496e+02,2.162273754403387557e+03,3.
  659841872412106568e+03,1.167389104634420846e+03,1.116522333940990393e+03,-7.
  243475142061594170e+02,1.154319035451656646e+03,1.494881545793028999e+04)
  b: T(0)=(4.694530637980111351e+03,2.101003495459988244e+03,3.
  207957317489257093e+03,1.393483476852222884e+04,3.629355746497698419e+03,-3.
  122852691247709117e+03,7.947428304515295167e+03,4.667399489592389727e+03,-1.
  305067346967445701e+03,-5.265219897161060203e+00,6.563177097183949627e+03,-1.
  114327671237384720e+01,3.737487417360972017e+02,2.074052821545472398e+03,3.
  957227271612303866e+03,1.778574118834093042e+03,1.504700200689648454e+03,1.
  784741275116079123e+03,1.626117516992070250e+03,1.463949266660182548e+04)
  T(1)=(4.884671151043686223e+03,2.016477673935037728e+03,2.
  563960072230776859e+03,8.422867877002190653e+03,3.971862617510039399e+03,-3.
  649419652202167981e+03,5.318344563287955680e+03,7.922373539234846248e+03,-2.
  036950370671189603e+03,-8.802316065768432054e+02,7.166888885237679460e+03,4.
  395276807519043359e+03,4.327169640156197943e+02,2.596848396524966574e+03,2.
  326512227834611167e+03,1.730263206899513989e+03,1.488983722766563687e+03,3.
  220665385083842921e+03,1.106893387043292933e+03,1.744565244349394925e+04)
  T(2)=(4.591014859591383356e+03,1.939544184320808199e+03,2.
  976166818405885351e+03,1.301132024216320497e+04,4.087264049876989702e+03,-3.
  398034635015794720e+03,6.939880618995275654e+03,8.703983955028777473e+02,-3.
  444151928813204449e+03,4.665937509833510376e+03,6.219091754047954055e+03,-7.
  533198432627855823e+00,7.007979966733214496e+02,2.162273754403384828e+03,3.
  659841872412107477e+03,1.167389104634420391e+03,1.116522333940990393e+03,-7.
  243475142061581664e+02,1.154319035451656873e+03,1.494881545793030273e+04)
  a: (4.591014859591384266e+03,1.939544184320808199e+03,2.976166818405885351e+
  03,1.301132024216321406e+04,4.087264049876992431e+03,-3.398034635015842468e+
  03,6.939880618995272016e+03,8.703983955028797936e+02,-3.444151928813199447e+
  03,4.665937509833509466e+03,6.219091754047957693e+03,-7.533198408782482147e+
  00,7.007979966733214496e+02,2.162273754403387557e+03,3.659841872412106568e+
  03,1.167389104634420846e+03,1.116522333940990393e+03,-7.243475142061594170e+
  02,1.154319035451656646e+03,1.494881545793028999e+04)
  b: (4.591014859591383356e+03,1.939544184320808199e+03,2.976166818405885351e+
  03,1.301132024216320497e+04,4.087264049876989702e+03,-3.398034635015794720e+
  03,6.939880618995275654e+03,8.703983955028777473e+02,-3.444151928813204449e+
  03,4.665937509833510376e+03,6.219091754047954055e+03,-7.533198432627855823e+
  00,7.007979966733214496e+02,2.162273754403384828e+03,3.659841872412107477e+
  03,1.167389104634420391e+03,1.116522333940990393e+03,-7.243475142061581664e+
  02,1.154319035451656873e+03,1.494881545793030273e+04)
  a: -7.533198408782482147e+00
  b: -7.533198432627855823e+00
```

**Comment by prayush** [comment]
This issue slipped under my radar somehow - I've pushed a fix for it with #1454 now.
